### PR TITLE
412 admin ban feed

### PR DESCRIPTION
**Pull request by ivanzaqqa**
#412 

### Breakdown:
- add ban feeds

### CODE REVIEW

File: app/routes/api.py [hunk 1]
```diff
@@ -756,6 +756,14 @@ def feeds(*args, **kwargs):
         socketio.emit('feeds', data.data.decode(), broadcast=True)
         return data
 
+@api.route('/feeds/<feed_id>/ban', methods=['DELETE'])
+@token_required
+def feeds_delete_at_banned(feed_id, *args, **kwargs):
+    user = kwargs['user'].as_dict()
+    if (request.method == 'DELETE'):
+        return FeedController.bannedfeeds(user, feed_id)
+    if (request.method == 'GET'):
```
Comment (mgufrone) [inline app/routes/api.py]: I'm no expert in python, but why do we need this definition? because on the above, I already see that the methods will be filtered based on that arguments.

File: migrations/versions/0b20a93243e4_create_feed_table.py [hunk 2]
```diff
@@ -25,7 +25,8 @@ def upgrade():
         sa.Column('message', sa.Text),
         sa.Column('attachment', sa.String(255)),
         sa.Column('created_at', sa.DateTime),
-        sa.Column('updated_at', sa.DateTime)
+        sa.Column('updated_at', sa.DateTime),
+        sa.Column('deleted_at',sa.DateTime)
```
Comment (mgufrone) [inline migrations/versions/0b20a93243e4_create_feed_table.py]: I think you're supposed to create a new migration instead of updating the existing migration. cc: @andy-shi88 @erdivartanovich

Comment (ivanzaqqa) [reply migrations/versions/0b20a93243e4_create_feed_table.py]: yesterday I've tried to make a new migration but can not because the table name is clashed mas

Comment (ivanzaqqa) [reply migrations/versions/0b20a93243e4_create_feed_table.py]: @mgufrone

Comment (andy-shi88) [reply migrations/versions/0b20a93243e4_create_feed_table.py]: yes @mgufrone , 
@ivanzaqqa  I think you were trying to create new table which caused errors there.

Comment (ivanzaqqa) [reply migrations/versions/0b20a93243e4_create_feed_table.py]: fixed mas @mgufrone @andy-shi88

File: app/routes/api.py (hunk 1 above)
Comment (andy-shi88) [reply app/routes/api.py]: if we only apply one verb on the route there is no need to add more condition @ivanzaqqa , pleas do read and understand the code

File: app/services/feed_service.py [hunk 3]
```diff
@@ -94,3 +94,14 @@ def save_file(self, file, id=None):
 			return current_app.config['SAVE_FEED_PHOTO_DEST'] + filename
 		else:
 			return None
+
+	def bannedfeeds(self, user, feed_id):
+		response = ResponseBuilder()
+		try:
+			feed = db.session.query(
```
Comment (andy-shi88) [inline app/services/feed_service.py]: ok, mind to explain this code? @ivanzaqqa

File: app/models/feed.py [hunk 4]
```diff
@@ -20,7 +20,10 @@ class Feed(db.Model, BaseModel):
 	user = db.relationship('User')
 	created_at = db.Column(db.DateTime)
 	updated_at = db.Column(db.DateTime)
+	deleted_at = db.Column(db.DateTime)
 
 	def __init__(self):
 		self.created_at = datetime.now() + timedelta(hours=7) 
 		self.updated_at = datetime.now() + timedelta(hours=7)
+		self.updated_at = datetime.now() + timedelta(hours=7)
```
Comment (andy-shi88) [inline app/models/feed.py]: what is this? @ivanzaqqa

File: app/models/feed.py [hunk 5]
```diff
@@ -20,7 +20,5 @@ class Feed(db.Model, BaseModel):
 	user = db.relationship('User')
 	created_at = db.Column(db.DateTime)
 	updated_at = db.Column(db.DateTime)
-
-	def __init__(self):
-		self.created_at = datetime.now() + timedelta(hours=7) 
-		self.updated_at = datetime.now() + timedelta(hours=7)
+	deleted_at = db.Column(db.DateTime)
```
Comment (andy-shi88) [inline app/models/feed.py]: attention to the indentation @ivanzaqqa , did you test your code?

File: app/models/feed.py [hunk 6]
```diff
@@ -20,7 +20,5 @@ class Feed(db.Model, BaseModel):
 	user = db.relationship('User')
 	created_at = db.Column(db.DateTime)
 	updated_at = db.Column(db.DateTime)
-
-	def __init__(self):
```
Comment (andy-shi88) [inline app/models/feed.py]: ok, here why did you remove the constructor? @ivanzaqqa

File: app/models/feed.py (hunk 4 above)
Comment (ivanzaqqa) [reply app/models/feed.py]: fixed @andy-shi88

File: app/routes/api.py [hunk 7]
```diff
@@ -756,6 +756,12 @@ def feeds(*args, **kwargs):
         socketio.emit('feeds', data.data.decode(), broadcast=True)
         return data
 
+@api.route('/feeds/<feed_id>/ban', methods=['DELETE'])
+@token_required
+def feeds_delete_at_banned(feed_id, *args, **kwargs):
+    user = kwargs['user'].as_dict()
+    if (request.method == 'DELETE'):
```
Comment (andy-shi88) [inline app/routes/api.py]: only one verb there, no need for any further check

Comment (ivanzaqqa) [reply app/routes/api.py]: fixed @andy-shi88

File: app/models/feed.py [hunk 8]
```diff
@@ -7,7 +7,7 @@ class Feed(db.Model, BaseModel):
 
 	__tablename__ = 'feeds'
 
-	visible = ['id', 'message', 'user_id', 'attachment', 'type', 'redirect_url', 'sponsor_id', 'created_at', 'updated_at']
```
Comment (andy-shi88) [inline app/models/feed.py]: you should have not deleeted this, be carefull when solving conflicts

File: app/models/feed.py [hunk 9]
```diff
@@ -23,8 +23,9 @@ class Feed(db.Model, BaseModel):
 	redirect_url = db.Column(db.String)
 	created_at = db.Column(db.DateTime)
 	updated_at = db.Column(db.DateTime)
+	deleted_at = db.Column(db.DateTime)
 
 	def __init__(self):
-		self.created_at = datetime.now() + timedelta(hours=7)
```
Comment (andy-shi88) [inline app/models/feed.py]: mind to explain why you deleted these lines?

File: app/models/feed.py (hunk 8 above)
Comment (ivanzaqqa) [reply app/models/feed.py]: in this file i add deleted_at mas, is it necessary? @andy-shi88

File: app/models/feed.py (hunk 9 above)
Comment (ivanzaqqa) [reply app/models/feed.py]: fixed @andy-shi88

File: app/models/feed.py (hunk 8 above)
Comment (andy-shi88) [reply app/models/feed.py]: hm, still, we need the redirect_url, type and sponsor id, you should keep those...;.

Comment (ivanzaqqa) [reply app/models/feed.py]: oh god, i'm sorry mas, I am not careful, i'm very sorry

**Comment by andy-shi88** [comment]
some code and additional condition check still need to be done, I'll merge this and handle it.
